app.js

```diff
@@ -13,6 +13,49 @@ function hasFilter(name, filters) {
   return false;
 }
 
+function stateToUrl(state) {
+  var qsParams = Object.keys(state);
+
+  if (!qsParams.length) {
+    return "";
+  }
+
+  return "?" + qsParams.map(function(param) {
+    return param + "=" + state[param];
+  }).join("&");
+}
+
+function initialSetStateByUrl() {
+  if (!location.search) {
+    return setDefaultState();
+  }
+
+  var state = location.search.replace("?", "").split('&')
```

Comment:
We could use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams here too